src/Microsoft.ML.Api/TypedCursor.cs

```diff
@@ -282,12 +282,7 @@ private Action<TRow> GenerateSetter(IRow input, int index, InternalSchemaDefinit
                     else if (fieldType.GetElementType() == typeof(bool))
                     {
                         Ch.Assert(colType.ItemType.IsBool);
-                        return CreateConvertingVBufferSetter<DvBool, bool>(input, index, poke, peek, x => (bool)x);
-                    }
-                    else if (fieldType.GetElementType() == typeof(bool?))
-                    {
-                        Ch.Assert(colType.ItemType.IsBool);
-                        return CreateConvertingVBufferSetter<DvBool, bool?>(input, index, poke, peek, x => (bool?)x);
+                        return CreateConvertingVBufferSetter<bool, bool>(input, index, poke, peek, x => x);
```

Comment:
fall back to  // VBuffer<T> -> T[]

Reply:
that is correct.

---
In reply to: [211109159](https://github.com/dotnet/machinelearning/pull/692#discussion_r211109159) [](ancestors = 211109159)